src/workload.rs

```diff
@@ -329,6 +329,8 @@ impl WorkloadInformation {
 #[derive(serde::Serialize, Default, Debug)]
 pub struct WorkloadStore {
     workloads: HashMap<IpAddr, Workload>,
+    // workload_to_vip maintains a mapping of workload IP to VIP. This is used only to handle removals.
+    workload_to_vip: HashMap<IpAddr, HashSet<(SocketAddr, u16)>>,
```

Comment:
Shall we consider the dual stack in advance? If yes, we may need to do reasonable design for it.

Reply:
What is the issue? IpAddr and SocketAddr can both be IPv4 or IPv6, right?

Reply:
I mean that a workload with IPv4 and IPv6 would have 2 entries in `workloads`, is that what we want?

Reply:
```
pub struct AdsClient {
    config: Config,
    /// Stores all known workload resources (by name)
    workloads: HashSet<String>,
...
```
As you can see, the workloads are identified by name in `AdsClient`, I think there may be an issue.

Reply:
Yes, we want 2 workloads in that case IMO. `name` is IP

Reply:
Well its a bit confusing. ResourceName == IP, but `name` is the pod name

Reply:
yes, shall I change it as?
```
workloads: HashSet<IpAddr>,
```

Reply:
Sorry, @howardjohn, I should do this change:
```
pub struct ResourceKey {
    pub resource_name: String,
    pub type_url: String,
}
```

Reply:
No it's good as is

Reply:
and 
```
pub struct XdsResource<T: prost::Message> {
    pub resource_name: String,
    pub resource: T,
}
```

Reply:
I'm not sure I understand what you mean

Reply:
> Well its a bit confusing. ResourceName == IP, but `name` is the pod name

I understand that everything is okay for now except there are some field name confusing in `ResourceKey` definition. 
If okay, no anything need to be changed here.